sszgen/main.go

```diff
@@ -221,14 +229,14 @@ func (v *Value) copy() *Value {
 type Type int
 
 const (
+	// TypeUndefined is a sentinel zero value to make initialization problems detectable
+	TypeUndefined Type = iota
```

Comment:
I added this new Type constant so that uninitialized Values using the zero-value for Type would cause failures. Previously Values would default to `uint` as their type, which could make for subtle errors. This enabled me to detect a few subtle parsing bugs while working on this branch and is generally a best practice when dealing with this kind of enum.